encoding/encoding.go

```diff
@@ -169,3 +172,31 @@ func (e rawSerializer) Response(res *transport.Response) (interface{}, error) {
 func (e rawSerializer) CheckSuccess(res *transport.Response) error {
 	return nil
 }
+
+// unspecifiedHealthSerializer is a lazy serializer that is used int the case of
+// unspecified encoding to determine the encoding type after transport
+// type has been determined.
+type unspecifiedHealthSerializer struct{}
+
+func (unspecifiedHealthSerializer) Encoding() Encoding { return UnspecifiedEncoding }
+
+func (unspecifiedHealthSerializer) Request(input []byte) (*transport.Request, error) {
+	return nil, errUnspecifiedHealthSerializer
+}
+
+func (unspecifiedHealthSerializer) Response(res *transport.Response) (interface{}, error) {
+	return nil, errUnspecifiedHealthSerializer
+}
+
+func (unspecifiedHealthSerializer) CheckSuccess(res *transport.Response) error {
+	return errUnspecifiedHealthSerializer
+}
+
+func (unspecifiedHealthSerializer) ThriftHealthSerializer() Serializer {
+	method, spec := getHealthSpec()
+	return &thriftSerializer{method, spec, defaultOpts}
+}
+
+func (unspecifiedHealthSerializer) ProtoHealthSerializer(serviceName string) Serializer {
```

Comment:
rather than pass the `serviceName` again here, can we store it in a field for `unspecifiedHealthSerializer` and use that instead